### PR TITLE
Modularise AppTP processors

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1202005414608929/f

### Description
See [asana](https://app.asana.com/0/488551667048375/1202005414608929/f) for full context.

### Steps to test this PR

_Smoke AppTP tests_
- [x] Ensure Device has android 12 installed
- [x] install from this branch, open and enable AppTP
- [x] verify browsing works normally
- [x] Disable and re-enable ApptP
- [x] verify browsing continues working as expected
- [x] Open apps like Twitter, instagram, etc. (ensure those apps are AppTP protected first)
- [x] verify apps work as expected
- [x] open apps that have trackers in them (ensure those apps are AppTP protected first)
- [x] verify apps work as expected and trackers are blocked

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #1959** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/1959" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/1959?utm_source=stack-comment).

File: vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/processor/VpnPacketInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        fun proceed(packetRequest: PacketRequest): Int
+
+        fun request(): PacketRequest
+    }
+}
+
+data class PacketRequest(
+    val packetInfo: PacketInfo,
+    val byteBuffer: ByteBuffer,
+    val byteChannel: ByteChannel,
```
Comment (joshliebe) [inline vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/processor/VpnPacketInterceptor.kt:73]: `ByteChannel` FTW 👍

File: vpn/src/main/java/com/duckduckgo/mobile/android/vpn/network/channels/NetworkChannelCreator.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    vpnServiceProvider: Provider<TrackerBlockingVpnService>
+) : NetworkChannelCreator {
+
+    private val vpnService by lazy { vpnServiceProvider.get() }
+
+    @Throws(IOException::class)
+    override fun createDatagramChannelAndConnect(inetSocketAddress: InetSocketAddress): DatagramChannel {
+        return DatagramChannel.open().also { channel ->
+            channel.configureBlocking(false)
+            channel.socket().let { socket ->
```
[the remark this answers is not in the document]
Comment (aitorvs) [reply vpn/src/main/java/com/duckduckgo/mobile/android/vpn/network/channels/NetworkChannelCreator.kt:63]: Good catch, no need for a let. I guess it was prob meant to be an `apply` back then. Will change it to that.

**Review by joshliebe — APPROVED** [review]
LGTM 👍

**Pull request merged**